frame/assets/src/functions.rs

```diff
@@ -205,7 +205,8 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		keep_alive: bool,
 	) -> Result<T::Balance, DispatchError> {
 		let details = Asset::<T, I>::get(id).ok_or(Error::<T, I>::Unknown)?;
-		ensure!(!details.is_frozen, Error::<T, I>::Frozen);
+		ensure!(details.status != AssetStatus::Destroying, Error::<T, I>::AssetNotLive);
```

Comment:
why not just details.status == AssetStatus::Live ?
If someone introduces `TempFrozen` for example, this might be an issue. 
there are few other cases like this below.

Reply:
Because it's dangerous if in future someone adds another status that this function should work for. When that happens, the person would have to remember to update all the extrinsics again.

Reply:
What about this?

Reply:
That's true, it skipped me. i've pushed an update